lab-jonathan/test/test-chat.js

```diff
@@ -0,0 +1,54 @@
+let server = require('server.js');
+let net = require('net');
+var assert = require('chai').assert;
+
+let nickname = require('server.js').nickname;
+//test server starts
+describe('test the server connection', function(){
+  before(function() {
+    server.listen(3000);
+  });
+  describe('server behavior', function(){
+    before(function(done){
+      let client;
+      client = net.createConnection({port: 3000});
+      done();
+    })
+    it('should start', function(done){
+      console.log('it started!' + client);
+      done();
+    })
+  })
+  //client stuff
+  describe('a client should be created', function(){
+    before(function(){
+
+    });
+    describe('client behavior', function(){
+      it('should be able to send a message to a single user', function(done){
+        expect(direct("hello").to.equal('hello'));
```

Comment:
Similar to my comment in your server.js, you should test the functionality of the server/lib, and should be able to do that without having for forcefully export methods and properties like `direct()`.